llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -1834,9 +1838,30 @@ bool TargetLowering::SimplifyDemandedBits(
 
       // Try shrinking the operation as long as the shift amount will still be
       // in range.
```

Comment:
This is now very similar to the `(shl i64:x, K) -> (i64 zero_extend (shl (i32 (trunc i64:x)), K))` fold below, this patch is a bugfix so no need to merge them yet, but maybe add a TODO?